### PR TITLE
Display template message

### PR DESCRIPTION
**Pull request by jhadvig**
Here is the first draft for the "Post-process template message" card. 
Wasn't sure if we want to also have possibility to edit the message in the `newfromtemplate` page, so I've add the extra input field just so we have the picture. Although not sure if it makes sense to update the message there(maybe if you don't save the template upon the creation process?!?).
screenshot-6

Here is also how we would display the message on the `next-step` page.
screenshot-7
Wasnt sure with the header wording. Wanted to go with 'Template Owner Message' , but went with 'Template Message'

@jwforres thoughts

### CODE REVIEW

**Comment by jhadvig** [comment]
will add generated `dist` after the review :)

File: app/views/create/next-steps.html [hunk 1]
```diff
@@ -89,6 +89,11 @@ <h1 ng-if="!pendingTasks(tasks()).length && erroredTasks(tasks()).length">Comple
               </div>
             </div>
 
+            <div ng-if="templateMessage.length">
+              <h2>Template message</h2>
+              <p>{{templateMessage}}</p>
```
Comment (liggitt) [inline app/views/create/next-steps.html]: I would expect whitespace to be preserved here, so they can have a couple paragraphs. I also think we should probably linkify the text (if they want to point to follow up docs)

File: app/views/newfromtemplate.html [hunk 2]
```diff
@@ -39,6 +39,19 @@
                               can-toggle="false"
                               help-text="Each label is applied to each created resource.">
                           </label-editor>
+                          <osc-form-section
+                            header="Message"
+                            about-title="message"
+                            about="Informative message for the user how to utilize the newly created resources. Message can accept parameters in '${PARAMETER}' format. Parameter substitution will be performed on the message before being displayed so that generated credentials and other parameters can be included in the output."
+                            expand="expand"
+                            can-toggle="canToggle">
+                              <input
```
Comment (liggitt) [inline app/views/newfromtemplate.html]: should probably be a textarea

**Comment by jwforres** [comment]
We should not be letting them edit that message, thats not the point of it.  It is from a template author as instructions to the user for after the template as been instantiated.

File: app/views/create/next-steps.html (hunk 1 above)
Comment (jwforres) [inline app/views/create/next-steps.html]: look at osc-image-summary.html  the div with class of resource-description, it preserves newlines and does a  linkify on the text

**Comment by jwforres** [comment]
the message from the template author is arguably the most important piece of information on the next steps page, seems like it should be the first thing they see.  This is what v2 looked like

v2-next-steps


Don't necessarily want the massive green block like we had there, but it was very clear that it was an important message to you that you should look at

cc @ajacobs21e @rhamilto

**Comment by jhadvig** [comment]
@jwforres here are some proposals for the template message on the next-step page
1# Similar to what we used in v2
screenshot-8
2# Same as 1# but with info icon and with the same similar font color
screenshot-12
screenshot-13


3#  Here I've played a bit with colors since we use the green one for success related messages, so picked blue and yellowish that bootstrap is using for warning and info alerts
screenshot-14
screenshot-15

4# And a color pallet that patternfly is using for info alerts
screenshot-18

Personally I like the blue one with black font color which looks really nice:
screenshot-17


cc'ing @spadgett && @sg00dwin for their opinion. Thanks ! :)

**Comment by jhadvig** [comment]
Also not really sure if "Message" is the right word here, since the real meaning is something like "ReadMe" or "Template instructions"

Thoughts ?

**Comment by jwforres** [comment]
I'm not sure we even need a header in the block, its text that the user doesn't really need, they just need the content of the message.

As far as the color palette my preference would be the blue with black text, the default patternfly info colors dont really work here since we have a gray background already.  But I would like an opinion from @ajacobs21e @sjcox-rh and @rhamilto

**Comment by spadgett** [comment]
Some of these options don't have enough color contrast for accessibility. See the "Contrast Ratios" here:

https://www.patternfly.org/styles/color-palette

Glancing at them, I prefer the blue background and black font.

**Comment by spadgett** [comment]
I'm not sure "Template Message" is the right header, but I don't have a good suggestion :(
Maybe just a (i) icon to the left and the message.

**Comment by jhadvig** [comment]
@jwforres @spadgett 
screenshot-16

**Comment by sjcox-rh** [comment]
Black text on blue works for me as well as long as this is the primary content you want the user to focus on while on this page.  I do like the idea of the (i) icon to the left of this message since you are leaning towards no header.  Not sure if the spacing would be thrown off when it's added though.  Worth a try!

**Comment by spadgett** [comment]
I was thinking of a large-ish info icon to the left, then a text block where the text left-aligns with the first line when it wraps.

**Comment by jhadvig** [comment]
@spadgett @sjcox-rh PTAL there three proposal based on you feed bac, thanks so much !
1#
screenshot-19
2#
screenshot-20
3#
screenshot-21

IMHO I really the first one.

**Comment by spadgett** [comment]
@jhadvig I was thinking of the first, except the text is left-aligned as a block. The text doesn't wrap under the icon.

**Comment by spadgett** [comment]
Although I think I'm OK with it as shown as well :)

**Comment by jhadvig** [comment]
@jwforres I've updated the PR based on the feedback from @spadgett and @sjcox-rh

**Comment by sjcox-rh** [comment]
@jhadvig Agree with @spadgett on the alignment of the text next to the icon in the first option!  Looks goods, otherwise!

File: app/scripts/controllers/create/nextSteps.js [hunk 3]
```diff
@@ -51,8 +51,10 @@ angular.module("openshiftConsole")
       }
     ];
 
-    $scope.parameters = ProcessedParametersService.getParams();
-    ProcessedParametersService.clearParams();
+    var processedTemplateData = ProcessedTemplateService.getTemplateData();
```
Comment (jwforres) [inline app/scripts/controllers/create/nextSteps.js]: make it return a map not an array, its less fragile

File: app/scripts/controllers/newfromtemplate.js [hunk 4]
```diff
@@ -143,7 +143,7 @@ angular.module('openshiftConsole')
               };
 
               if (!_.isEmpty(config.parameters)) {
```
Comment (jwforres) [inline app/scripts/controllers/newfromtemplate.js]: dont rely no the existence of config.parameters in order to set config.message, a template author may have a message but no parameters

**Comment by jhadvig** [comment]
@jwforres comments addressed. PTAL :)

**Comment by jwforres** [comment]
[merge]

**Comment by jwforres** [comment]
@bparees FYI console changes are going in

**Comment by bparees** [comment]
awesome!

**Pull request merged**